### ISSUE TITLE
A few issues using the `decapoda-research/llama-7b-hf` model as hf config seems to have changed

### ISSUE THREAD

**Issue by mathieul** [elixir-nx/bumblebee#225]
I am still testing the models from Sears Moriarty's  [Open-Source Elixir Alternatives to ChatGPT](https://dockyard.com/blog/2023/05/16/open-source-elixir-alternatives-to-chatgpt) post. I'm having some issues with the `decapoda-research/llama-7b-hf` model.

Here's the code:

```elixir
{:ok, model} = Bumblebee.load_model({:hf, "decapoda-research/llama-7b-hf"})
{:ok, tokenizer} = Bumblebee.load_tokenizer({:hf, "decapoda-research/llama-7b-hf"})
{:ok, generation_config} = Bumblebee.load_generation_config({:hf, "decapoda-research/llama-7b-hf"})
serving = Bumblebee.Text.generation(model, tokenizer, generation_config, defn_options: [compiler: EXLA])
Nx.Serving.run(serving, "Elixir is a")
```

A few things seem to have changed on huggingface:

1. loading the model fails because the architecture returned from the spec is `LLaMAForCausalLM` (https://huggingface.co/decapoda-research/llama-7b-hf/blob/main/config.json) and Bumblebee expects it to be `LlamaForCausalLM` (case difference - https://github.com/elixir-nx/bumblebee/blob/main/lib/bumblebee.ex#L133)
2. once this fixed, loading the tokenizer fails because its name is `tokenizer_config.json` (https://huggingface.co/decapoda-research/llama-7b-hf/blob/main/tokenizer_config.json) and Bumblebee expects `tokenizer.json` 
3. once this fixed, the `tokenizers` library fails to parse it and returns this error:

```elixir
iex(1)> {:ok, tokenizer} = Bumblebee.load_tokenizer({:hf, "decapoda-research/llama-7b-hf"})
** (RuntimeError) failed to read tokenizer from file, reason: Tokenizer(Error("expected `,` or `}`", line: 1, column: 13))
    (bumblebee 0.3.0) lib/bumblebee/utils/tokenizers.ex:146: Bumblebee.Utils.Tokenizers.load!/1
    (bumblebee 0.3.0) lib/bumblebee/text/llama_tokenizer.ex:8: Bumblebee.HuggingFace.Transformers.Config.Bumblebee.Text.LlamaTokenizer.load/2
    (bumblebee 0.3.0) lib/bumblebee.ex:741: Bumblebee.load_tokenizer/2
```

And here's the tokenizer config content (https://huggingface.co/decapoda-research/llama-7b-hf/resolve/main/tokenizer_config.json):

```json
{"bos_token": "", "eos_token": "", "model_max_length": 1000000000000000019884624838656, "tokenizer_class": "LLaMATokenizer", "unk_token": ""}
```

I imagine those changes are part of a bigger one, does someone knows what it is?

**Comment by jonatanklosko** [comment]
Hey! The `decapoda-research/llama-7b-hf` repo is a bit broken honestly, it was generated before the llama PR was finished in huggingface/transformers, hence the capitalization difference. As for the tokenizer there are generally two variants: slow (using `tokenizer_config.json` and fast using `tokenizer.json`), oftentimes a specific repo doesn't have the fast `tokenizer.json`, but it's possible to find a different repo with the same tokenizer that has it.

tl;dr try `huggyllama/llama-7b`, as far as I remember it is the same checkpoint as `decapoda-research/llama-7b-hf`, but shouldn't have any of these issues :)
